src/libraries/System.Memory.Data/src/System/BinaryData.cs

```diff
@@ -310,7 +302,18 @@ public override unsafe string ToString()
         /// <param name="jsonTypeInfo">The <see cref="JsonTypeInfo"/> to use when serializing to JSON.</param>
         /// <returns>The data converted to the specified type.</returns>
         public T? ToObjectFromJson<T>(JsonTypeInfo<T> jsonTypeInfo)
-            => JsonSerializer.Deserialize<T>(_bytes.Span, jsonTypeInfo);
+            => JsonSerializer.Deserialize(GetBytesWithTrimmedBom(), jsonTypeInfo);
+
+        private ReadOnlySpan<byte> GetBytesWithTrimmedBom()
+        {
+            ReadOnlySpan<byte> span = _bytes.Span;
+
+            // Check for the UTF-8 byte order mark (BOM) EF BB BF
+            if (span.Length > 2 && span[0] == 0xEF && span[1] == 0xBB && span[2] == 0xBF)
+                span = span.Slice(3);
```

Comment:
Might want to use a pattern if you're feeling particularly fancy? 😄
```suggestion
            if (span is [0xEF, 0xBB, 0xBF, ..ReadOnlySpan<byte> tail])
                span = tail;
```

Reply:
The library also targets netstandard and we can't use polysharp unfortunately :-(